src/parser.rs

```diff
@@ -57,8 +65,24 @@ pub async fn parse_fragment_async<R: AsyncRead + Unpin>(
 }
 
 pub async fn parse_document_async<R: AsyncRead + Unpin>(r: R) -> io::Result<Handle> {
-    let parser = driver::parse_document(RcDom::default(), Default::default());
-    parse_internal_async(parser, r).await
+    let parser = driver::parse_document(RcDomWithLineNumbers::default(), create_error_opts());
+    let dom = parse_internal_async(parser, r).await?;
+    dom.create_error_from_parse_errors()?;
+
+    Ok(dom.document().clone())
+}
+
+fn create_error_opts() -> ParseOpts {
+    ParseOpts {
+        tokenizer: TokenizerOpts {
+            exact_errors: true,
+            ..Default::default()
+        },
+        tree_builder: TreeBuilderOpts {
+            exact_errors: true,
+            ..Default::default()
+        },
+    }
```

Comment:
Why is this suddenly needed? The OP says:

> Previously they were being stored in the RcDom instance's errors vector, and ignored.

This reads as if RcDom was generating errors this whole time (without having any `ParseOpts` that specify exposure of errors), and now all we need to do is start paying attention to them, instead of introducing a new `ParseOpts` config argument. I'm assuming that's not the case though?